src/common/transformations/src/transformations/common_optimizations/matmul_split_decomposition.cpp

```diff
@@ -0,0 +1,214 @@
+
+    auto bias_pattern = wrap_type<opset1::Constant>();
+    auto add_pattern = wrap_type<opset1::Add>({matmul_pattern, bias_pattern});
+
+    auto reshape_productor_pattern = std::make_shared<pattern::op::Or>(OutputVector{matmul_pattern, add_pattern});
+
+    auto reshape_pattern = wrap_type<opset1::Reshape>({reshape_productor_pattern, any_input()});
+    auto transpose_pattern = wrap_type<opset6::Transpose>({reshape_pattern, any_input()});
+    auto reshape2_pattern = wrap_type<opset1::Reshape>({reshape_pattern, any_input()});
+
+    auto reshape_or_transpose_pattern =
+        std::make_shared<pattern::op::Or>(OutputVector{reshape2_pattern, transpose_pattern});
+
+    matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        // Heuristics: there should be only 3 gathers to split
+        auto root_node = m.get_match_root();
+        bool have_transpose = as_type<opset1::Transpose>(root_node.get()) != nullptr;
+        auto children = root_node->get_output_target_inputs(0);
+        if (children.size() != 3u) {
+            return false;
+        }
+
+        auto matmul = pattern_map.at(matmul_pattern).get_node_shared_ptr();
+        auto weights = matmul->input_value(1);
+        std::shared_ptr<ov::Node> add = nullptr;
+        bool have_bias = false;
+        for (auto& consumer : matmul->get_output_target_inputs(0)) {
+            if (ov::is_type<opset1::Add>(consumer.get_node()->shared_from_this())) {
+                add = pattern_map.at(add_pattern).get_node_shared_ptr();
+                have_bias = true;
+                break;
+            }
+        }
+        const bool& transpose_b = as_type_ptr<opset1::MatMul>(matmul)->get_transpose_b();
+        const auto& reshape = pattern_map.at(reshape_pattern);
+        auto concat = reshape.get_node_shared_ptr()->input_value(1);
+
+        auto reshape_output_shape_rank = reshape.get_node_shared_ptr()->get_output_partial_shape(0).rank().get_length();
+        if (reshape_output_shape_rank != 5) {
+            // Baichun also match this pattern, but it only has rank 4, and have performance regression, so filter it
+            // out.
+            return false;
+        }
+
+        NodeVector gathers, fake_quantizes;
+        gathers.resize(3);
+        fake_quantizes.resize(3);
+        for (auto& child : children) {
+            std::shared_ptr<ov::Node> fq = nullptr;
+            auto gather = child.get_node()->shared_from_this();
+            if (ov::is_type<opset1::FakeQuantize>(gather)) {
+                fq = gather;
+                gather = gather->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+            }
+            if (ov::is_type<ov::op::util::GatherBase>(gather)) {
+                const auto axis_node = as_type_ptr<opset6::Constant>(gather->input_value(2).get_node_shared_ptr());
```

Comment:
Minor: `input_value(i).get_node_shared_ptr()` can be replaced with shorter code: `get_input_node_shared_ptr(i)`. But I don't insist on these changes